discovery/gossiper.go

```diff
@@ -1231,13 +1233,46 @@ func (d *AuthenticatedGossiper) retransmitStaleChannels(now time.Time) error {
 		signedUpdates = append(signedUpdates, chanUpdate)
 	}
 
-	// If we don't have any channels to re-broadcast, then we'll exit
+	// We'll also check that our NodeAnnouncement is not too old.
+	currentNodeAnn, err := d.cfg.SelfNodeAnnouncement(false)
+	if err != nil {
+		return fmt.Errorf("unable to get current node announment: %v",
+			err)
+	}
+
+	timestamp := time.Unix(int64(currentNodeAnn.Timestamp), 0)
```

Comment:
Just something to note, the timestamp of our node announcement is updated to `time.Now` every time the server is started.

Reply:
Because of this, if a node restarts before it reaches a full day of uptime, it'll never rebroadcast its node announcement.

Reply:
I don't think that's the case. We only seem to send when opening a channel or when running with `--nat`.